File: app/src/main/java/vergecurrency/vergewallet/service/model/wallet/CreateAddressErrorResponse.java
package vergecurrency.vergewallet.service.model.wallet;

public class CreateAddressErrorResponse {

	enum Error {
		MAIN_ADDRESS_GAP_REACHED
	}

	private String code;
	private String message;

	public String getCode() {
		return code;
	}

	public void setCode(String code) {
		this.code = code;
	}

	public String getMessage() {
		return message;
	}

	public void setMessage(String message) {
		this.message = message;
	}

}

File: app/src/main/kotlin/vergecurrency/vergewallet/service/model/wallet/CreateAddressErrorResponse.kt
package vergecurrency.vergewallet.service.model.wallet

import com.google.gson.Gson

class CreateAddressErrorResponse {

    var code: String? = null
    var message: String? = null

    internal enum class Error {
        MAIN_ADDRESS_GAP_REACHED
    }

    companion object {

        fun decode(JSON: String): CreateAddressErrorResponse {
            return Gson().fromJson(JSON, CreateAddressErrorResponse::class.java)
        }
    }

}
